src/partition.c

```diff
@@ -1349,7 +1364,339 @@ static void pick_metis(int nodeID, struct space *s, int nregions,
+        } else {
+          weights_v[k] = 0;
+        }
+      }
+
+#ifdef SWIFT_DEBUG_CHECKS
+      /* Check weights are all in range. */
+      int failed = 0;
+      for (int k = 0; k < ncells; k++) {
+        if ((SCOTCH_Num)vertexw[k] < 0) {
+          message("Input vertex weight out of range: %ld", (long)vertexw[k]);
+          failed++;
+        }
+        if (weights_v[k] < 0) {
+          message("Used vertex weight  out of range: %ld", (long)weights_v[k]);
+          failed++;
+        }
+      }
+      if (failed > 0) error("%d vertex weights are out of range", failed);
+#endif
+    }
+
+    /* Init the edges weights array. */
+
+    if (edgew != NULL) {
+      for (int k = 0; k < 26 * ncells; k++) {
+        if (edgew[k] > 1) {
+          weights_e[k] = edgew[k];
+        } else {
+          weights_e[k] = 1;
+        }
+      }
+
+#ifdef SWIFT_DEBUG_CHECKS
+      /* Check weights are all in range. */
+      int failed = 0;
+      for (int k = 0; k < 26 * ncells; k++) {
+
+        if ((SCOTCH_Num)edgew[k] < 0) {
+          message("Input edge weight out of range: %ld", (long)edgew[k]);
+          failed++;
+        }
+        if (weights_e[k] < 1) {
+          message("Used edge weight out of range: %" "I64d", (long long)weights_e[k]);
+          failed++;
+        }
+      }
+      if (failed > 0) error("%d edge weights are out of range", failed);
+#endif
+    }
+    /* Define the cell graph. Keeping the edge weights association. */
+    int nadjcny = 0;
+    int nxadj = 0;
+    graph_init_scotch(s, s->periodic, weights_e, adjncy, &nadjcny, xadj,
+                      &nxadj);
+    /* Define the cell graph. Keeping the edge weights association. */
+    // Setting up the Scotch graph
+    SCOTCH_Graph graph;
+    SCOTCH_Num baseval = 0;
+    SCOTCH_Num vertnbr = ncells;         /* Number of vertices */
+    SCOTCH_Num edgenbr = (26 * vertnbr); /* Number of edges (arcs)   */
+
+    SCOTCH_Num *verttab; /* Vertex array [vertnbr + 1] */
+    if ((verttab = (SCOTCH_Num *)malloc((vertnbr + 1) * sizeof(SCOTCH_Num))) == NULL) {
+      error("Failed to allocate Vertex array");
+    }
+
+    SCOTCH_Num *velotab; /* Vertex load array        */
+    if ((velotab = (SCOTCH_Num *)malloc((vertnbr) * sizeof(SCOTCH_Num))) == NULL) {
+      error("Failed to allocate Vertex load array");
+    }
+
+    SCOTCH_Num *edgetab; /* Edge array [edgenbr]     */
+    if ((edgetab = (SCOTCH_Num *)malloc((edgenbr) * sizeof(SCOTCH_Num))) == NULL) {
+      error("Failed to allocate Edge array");
+    }
+
+    SCOTCH_Num *edlotab; /* Int load of each edge     */
+    if ((edlotab = (SCOTCH_Num *)malloc((edgenbr) * sizeof(SCOTCH_Num))) == NULL) {
+      error("Failed to allocate Edge Load array");
+    }
+
+    int edges_deg = 26;
+    for (int i = 0; i < vertnbr; i++) {
+      verttab[i] = i * edges_deg;
+      velotab[i] = weights_v[i];
+    }
+    verttab[vertnbr] = vertnbr * edges_deg;
+
+    int vertex_count = 0;
+    int neighbour;
+    int return_edge;
```

Comment:
GCC wants `return_edge` initialising. Best to keep it happy.

Reply:
Agreed. It is first used inside an if, so ... OK.